osi_vehicle.proto

```diff
@@ -431,6 +439,73 @@ message VehicleClass
         // Intentions, requests or wishes from the function to change the vehicle-state.
         //
         optional Intentions intentions = 4;
+        
+        // The possible functions.
+        // References: 
+        // - https://www.vda.de/en/topics/innovation-and-technology/automated-driving/automated-driving
+        //
+        enum FunctionName
+        {
+            // The function name is unknown.
+            //
+            FUNCTION_NAME_UNKNOWN = 0;
+            
+            // It is another function.
+            //
+            FUNCTION_NAME_OTHER = 1;
+            
+            // Autopilot in urban areas.
+            //
+            FUNCTION_NAME_LEVEL_4_URBAN_DRIVING = 2;
+            
+            // Driverless parking.
+            //
+            FUNCTION_NAME_LEVEL_4_VALET_PARKING = 3;
+            
+            // Autopilot on the highway.
+            //
+            FUNCTION_NAME_LEVEL_3_HIGHWAY_DRIVING = 4;
+            
+            // Autopilot for traffic jams.
+            //
+            FUNCTION_NAME_LEVEL_3_TRAFFIC_JAM_DRIVING = 5;
+            
+            // Parking done by function lateral and longitudinal.
+            //
+            FUNCTION_NAME_LEVEL_2_PARKING_ASSISTANT = 6;
+            
+            // Parking done by function lateral.
+            //
+            FUNCTION_NAME_LEVEL_1_PARKING_STEERING_ASSISTANT = 7;
+            
+            // Longitudinal control with distance keeping by function.
+            //
+            FUNCTION_NAME_LEVEL_1_ADAPTIVE_CRUISE_CONTROL = 8;
+            
+            // Lateral control by function to keep the lane.
+            //
+            FUNCTION_NAME_LEVEL_1_LANE_KEEP_ASSISTANT = 9;
+                        
+            // Longitudinal control by function.
+            //
+            FUNCTION_NAME_LEVEL_1_CRUISE_CONTROL = 10;
+            
+            // The maximum speed is limited by a function.
+            //
+            FUNCTION_NAME_LEVEL_1_LIMIT = 11;
+            
+            // Warning when the vehicle drives over the lane border markings.
+            //
+            FUNCTION_NAME_LEVEL_0_LANE_DEPARTURE_WARNING = 12;
+            
+            // Monitoring of the blind spot.
+            //
+            FUNCTION_NAME_LEVEL_0_BLIND_SPOT_MONITORING = 13;
+            
+            // Warning by the danger of a collision.
+            //
+            FUNCTION_NAME_LEVEL_0_FORWARD_COLLISION_WARNING = 14;
+        }
```

Comment:
Well done, thats a great list. Is it accounted for that some of these funtions can occur in parallel?

Reply:
localization has some  overlap with position and position_rmse in osi_hostvehicledata.proto